BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java

```diff
@@ -135,39 +89,61 @@ public BraintreeClient(@NonNull Context context, @NonNull String authorization,
      * @param returnUrlScheme     A custom return url to use for browser and app switching
      */
     public BraintreeClient(@NonNull Context context, @NonNull ClientTokenProvider clientTokenProvider, @NonNull String returnUrlScheme) {
-        this(createDefaultParams(context, null, clientTokenProvider, returnUrlScheme));
+        this(
+                new BraintreeOptions(context, IntegrationType.CUSTOM)
+                        .clientTokenProvider(clientTokenProvider)
+                        .returnUrlScheme(returnUrlScheme)
+        );
     }
 
     BraintreeClient(@NonNull Context context, @NonNull ClientTokenProvider clientTokenProvider, @NonNull String sessionId, @NonNull @IntegrationType.Integration String integrationType) {
-        this(createDefaultParams(context, null, clientTokenProvider, sessionId, integrationType));
+        this(
+                new BraintreeOptions(context, integrationType)
+                        .clientTokenProvider(clientTokenProvider)
+                        .sessionId(sessionId)
+        );
     }
 
     BraintreeClient(@NonNull Context context, @NonNull String authorization, @NonNull String sessionId, @NonNull @IntegrationType.Integration String integrationType) {
-        this(createDefaultParams(context, authorization, null, sessionId, integrationType));
+        this(
+                new BraintreeOptions(context, integrationType)
+                        .authorization(authorization)
+                        .sessionId(sessionId)
+        );
+    }
+
+    // NEXT MAJOR VERSION: Make BraintreeClient(BraintreeOptions) the primary constructor to
+    // allow additional allow parameters to grow over time without having to support multiple
+    // different constructors
```

Comment:
do we think this is going to be necessary once we move this to `Kotlin`? Optional parameters would fix all of this

Reply:
Yeah you're right. This was some cleanup related to a DropIn issue, but it's internal. We can probably close this PR.